option cc_generic_services = true;

message Empty { }

message Key {
	required string key = 1;
}

message Coords {
	required int32 x = 1;
	required int32 y = 2;
}

message Item {
	required int32 id = 1;
	required int32 count = 2;
}

message Inventory {
	repeated Item items = 1;
}

message PlayerInfo {
	optional int32 x = 1;
	optional int32 y = 2;
	optional int32 z = 3;
	optional int32 hp = 4;
	optional float angle = 5;
}

service CoreApi {
	rpc GetPlayerInfo(Empty) returns (PlayerInfo);
	rpc GetInventory(Empty) returns (Inventory);
	rpc GoToXY(Coords) returns (Empty);
	rpc MouseClick(Coords) returns (Empty);
	rpc PressKey(Key) returns (Empty);  
}
